cookbooks/aws-parallelcluster-common/resources/network_service/partial/_restart_network_debian.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+action :restart_network do
+  network_service_name = 'systemd-resolved'
+  Chef::Log.info("Restarting '#{network_service_name}' service, platform #{node['platform']} '#{node['platform_version']}'")
+
+  service network_service_name.to_s do
+    action %i(restart)
+    ignore_failure true
+  end
+end
+
+action :apply_network_changes do
+  action_restart_network
+
+  ruby_block "apply network configuration" do
```

Comment:
Nitpick. Is there a reason not to use Chef [execute](https://docs.chef.io/resources/execute/) resource?

Reply:
I just moved the already existing code. I can try to change it.

Reply:
changed to use execute, tested on ubuntu20